### ISSUE TITLE
URLs in sent mail with SPA redirect behavior and redirect host set

### ISSUE THREAD

**Issue by trilader** [Flask-Middleware/flask-security#487]
Hi.

I'm having issues with the links in emails not being correct (in my development setup).

I'm using SECURITY_REDIRECT_MODE="spa" together with SECURITY_REDIRECT_HOST="localhost:3000", the regular Flask development server is running at localhost:5000.

The email address confirmation email contains `http://localhost:5000/auth/confirm/<token>` as the confirmation link, which works as the Flask development server can handle it with the form based views from Flask-Security. I think it should go to the front end app on port 3000.

I'm also not sure how to handle this if the backend application will not run on the same host (or with a proxy in front) in production once the app is ready.

Am I missing some configuration option or is SECURITY_REDIRECT_HOST out of scope for this case?

Cheers
trilader

**Comment by jwag956** [comment]
The configuration variable is 'SECURITY_REDIRECT_BEHAVIOR'="spa".

Not quite sure about your second question - it is pretty much standard that in production there is a single server IP - usually fronted with nginx or some API gateway.

**Comment by trilader** [comment]
Not sure where I got the SECURITY_REDIRECT_MODE from (likely me typing it in instead of copy/paste) but I have/had SECURITY_REDIRECT_BEHAVIOR="spa" in my configuration.

Regarding the production thing: Okay nginx will take care of it.

Will that also resolve the email link issue? Both frontend and backend will have the same external address/name then.
Is there something (besides setting up nginx or something similar on the development machine) I can do to not have to edit the port in the URL when copy/pasting into the browser for development purposes?

**Comment by jwag956** [comment]
Ok - so yes - email links don't use REDIRECT_HOST - not sure why you would want your UI to get that GET request?
How it should work is that the email confirmation link will perform a GET to your backend - which will then redirect
to SECURITY_POST_CONFIRM_VIEW which will honor REDIRECT_HOST.
You need to set your POST_CONFIRM_VIEW to something your UI router has set up.

**Comment by trilader** [comment]
Ok. I'll have a look at that tomorrow. Thanks for your fast help.
